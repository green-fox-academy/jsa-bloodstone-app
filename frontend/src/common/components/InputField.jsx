import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native';
import CardView from './CardView';

const style = {
  height: 40,
  margin: 5,
  paddingStart: 10,
  fontSize: 14,
};

function InputField({
  value, onChangeText, placeholder, secureTextEntry,
}) {
  const [isFocus, setFocus] = useState(false);

  const extraStyle = {
    margin: 0,
    padding: 0,
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,.12)',
    elevation: isFocus ? 2 : 0,
  };

  return (
    <CardView style={extraStyle}>
      <TextInput
        style={style}
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        secureTextEntry={secureTextEntry}
        placeholderTextColor="#aaa"
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
      />
    </CardView>
  );
}

InputField.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
  secureTextEntry: PropTypes.bool,
};

InputField.defaultProps = {
  value: '',
  placeholder: 'placeholder',
  onChangeText: null,
  secureTextEntry: false,
};

export default InputField;
