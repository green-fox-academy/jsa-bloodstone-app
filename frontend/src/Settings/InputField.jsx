import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { TextInput, StyleSheet } from 'react-native';
import { CardView } from '../common/components';

function InputField({
  value, onChangeText, placeholder, secureTextEntry,
}) {
  const [isFocus, setFocus] = useState(false);

  const styles = StyleSheet.create({
    cardView: {
      width: 240,
      paddingVertical: 6,
      marginBottom: 0,
      borderWidth: 1,
      borderColor: 'rgba(0,0,0,.12)',
      elevation: isFocus ? 2 : 0,
    },
    textInput: {
      fontSize: 20,
    },
  });

  return (
    <CardView style={styles.cardView}>
      <TextInput
        style={styles.textInput}
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        secureTextEntry={secureTextEntry}
        placeholderTextColor="#aaa"
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
      />
    </CardView>
  );
}

InputField.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
  secureTextEntry: PropTypes.bool,
};

InputField.defaultProps = {
  value: '',
  placeholder: 'placeholder',
  onChangeText: null,
  secureTextEntry: false,
};

export default InputField;
