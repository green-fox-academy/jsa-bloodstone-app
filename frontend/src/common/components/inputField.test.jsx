import React from 'react';
import renderer from 'react-test-renderer';

import InputField from './InputField';

describe('<InputField />', () => {
  it('Should render input field and match the snapshot', () => {
    const component = renderer.create(<InputField />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
