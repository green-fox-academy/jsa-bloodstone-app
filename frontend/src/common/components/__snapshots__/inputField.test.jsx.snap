// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> Should render buildings and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 6,
        "elevation": 3,
        "marginBottom": 10,
        "padding": 15,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "x": 0,
          "y": 4,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      Object {
        "borderColor": "rgba(0,0,0,.12)",
        "borderWidth": 1,
        "elevation": 0,
        "margin": 0,
        "padding": 0,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={null}
    onFocus={[Function]}
    placeholder="placeholder"
    placeholderTextColor="#aaa"
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Object {
        "fontSize": 14,
        "height": 40,
        "margin": 5,
        "paddingStart": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
